### ISSUE TITLE
@workspace @workspace How would you rework the chats.js code ( copied from another project , so it's unusable like this) in our project ? For exemple I would like better separating of the js scripts.    Exemple, the activty report sample, I would like a js script only in charge of producing the "FiatBalanceHistoryChart"  ( it's generateBalanceHistoryChart on the chart.js old script )  How would you handle this for this new project ? For exemple "scripts/js/fiat_balance_history_chart.js"   One thing i thought, but maybe you have a better idea, is that when processing a new generation, we use a uuid for the whole generation. That way any file created by chartjs, etc will be using this id. At the end of the generation of the pdf ( even if there is an error, ) we can delete all the .temp files using that id

### ISSUE THREAD

**Issue by Qualzz** [microsoft/vscode-copilot-release#2245]
Type: <b>Bug</b>

Enable Claude 3.5 Sonnet (Preview) for all clients

It's asking me this for every chat, it loose context of the files after that ENABLE button

Extension version: 0.22.1
VS Code version: Code 1.95.1 (65edc4939843c90c34d61f4ce11704f09d3e5cb6, 2024-10-31T05:14:54.222Z)
OS version: Windows_NT x64 10.0.22621
Modes:
Connection to 'wsl+ubuntu' could not be established
Remote OS version: Linux x64 5.15.153.1-microsoft-standard-WSL2

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-10900K CPU @ 3.70GHz (20 x 3696)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|127.91GB (90.67GB free)|
|Process Argv|--open-url --crash-reporter-id 7834d3de-7015-4333-a6df-d461bbd01541 -- vscode://ms-vsliveshare.vsliveshare/join?vslsLink=https://prod.liveshare.vsengsaas.visualstudio.com/join%253F538527E2C4F5C0A97B06A14DFFC06CC8750A|
|Screen Reader|no|
|VM|20%|

Connection to 'wsl+ubuntu' could not be established

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 5.15.153.1-microsoft-standard-WSL2|
|CPUs|Intel(R) Core(TM) i9-10900K CPU @ 3.70GHz (20 x 0)|
|Memory (System)|62.76GB (58.14GB free)|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
pythonnoceb:30805159
asynctok:30898717
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
01bff139:31013167
dvdeprecation:31068756
dwnewjupyter:31046869
impr_priority:31102340
nativerepl1:31139838
refactort:31108082
pythonrstrctxt:31112756
cf971741:31144450
iacca1:31171482
notype1:31157159
5fd0e150:31155592
dwcopilot:31170013

```

</details>

**Comment by Qualzz** [comment]
Oh wow, I kinda don't like copilot to publish the issue with all my private content as a title. That was unexpected.

**Comment by lramos15** [comment]
Hmm that's not expected. Are you saying even after restarting VS Code you see it ask you to enable the policy every time?

**Comment by Qualzz** [comment]
The policy prompt kept appearing repeatedly, despite being enabled in my Copilot settings here: https://github.com/settings/copilot.

After restarting VS Code, Claude now seems unresponsive, stalling on "Generating....", so I can’t verify if the policy issue is still there. However, I suspect it may be resolved, as the policy acceptance message did appear before any interaction with Claude.

That said, a full VS Code restart shouldn’t be required after clicking "ENABLE" just once.

**Comment by lramos15** [comment]
> That said, a full VS Code restart shouldn’t be required after clicking "ENABLE" just once.

You are correct it should not, I will see if I can reproduce it but haven't been able to yet on my end. I was trying to offer a solution to the issue you were experiencing. 

Did you enable the policy initially from within VS code or from the website?

**Comment by Qualzz** [comment]
Sorry about the tone of that sentence, it's seems chatGPT is intentionally rewriting my answer in a less friendly tone than my messed up English.

I enabled it from within VS Code. 
Timeline:

click on Copilot Edits > select model > see that there is now Claude. > clicking on Claude > writing my message, adding context > send

Policy message from vs code > Click enable > Answer from Claude ( but it's unaware of any context, as if the policy message removed any contextual information/linked files )   

So I clicked on the + to create a new chat, and the whole exact same thing happen. 

Then I've restarted, and tried Claude again, but this time I have this stalling issue. 
And as of now, instead of a stale : ![Image](https://github.com/user-attachments/assets/424df462-7a18-4153-b2d4-69d1ff3396be)

**Comment by lramos15** [comment]
If you paste that request Id here I can look into what is happening. Thank you for the bug report. I will keep it open and try to reproduce.
